aws/solutions/DirectoryServiceSettings/src/directory_settings_custom_resource.py

```diff
@@ -0,0 +1,233 @@
+        for directory in page["DirectoryDescriptions"]:
+            return directory["Alias"], directory["SsoEnabled"]
+    return None
+
+
+def get_registered_topics(directory_id: str) -> List[str]:
+    """Get existing list of SNS topics configured for directory monitoring.
+
+    Args:
+        directory_id: Directory ID
+
+    Returns:
+        Existing SNS Topics configured for directory monitoring
+    """
+    response = ds_client.describe_event_topics(DirectoryId=directory_id)
+    registered_topics: list = []
+    for registered_topic in response["EventTopics"]:
+        registered_topics.append(registered_topic["TopicName"])
+    return registered_topics
+
+
+def register_directory_monitoring_topic(directory_id: str, topic: str):
+    """Register SNS topic for directory monitoring.
+
+    Args:
+        directory_id: Directory ID
+        topic: SNS topic name
+    """
+    registered_topics: list = get_registered_topics(directory_id)
+    if topic not in registered_topics:
+        response = ds_client.register_event_topic(DirectoryId=directory_id, TopicName=topic)
+        logger.info(f"Directory Monitoring registered with Topic '{topic}'")
+        logger.debug(f"register_topic_response = {json.dumps(response, default=str)}")
+
+
+def deregister_directory_monitoring_topic(directory_id: str, topic: str):
+    """Deregister SNS topic for directory monitoring.
+
+    Args:
+        directory_id: Directory ID
+        topic: SNS topic name
+    """
+    registered_topics: list = get_registered_topics(directory_id)
+    for registered_topic in registered_topics:
+        response = ds_client.deregister_event_topic(DirectoryId=directory_id, TopicName=registered_topic)
+        logger.info(f"Directory Monitoring deregistered with Topic '{registered_topic}'")
+        logger.debug(f"deregister_topic_response = {json.dumps(response, default=str)}")
+
+
+def create_directory_alias(directory_id: str, alias: str, existing_alias: str) -> str:
+    """Create and assigns an alias to the directory.
+
+    Args:
+        directory_id: Directory ID
+        alias: Requested directory alias
+        existing_alias: Existing directory alias
+
+    Raises:
+        Exception: Directory already has a different alias. Use existing alias for the 'DirectoryAlias' CloudFormation parameter.
+
+    Returns:
+        The directory alias, either the new alias created, or the existing alias that was already configured.
+    """
+    if alias == existing_alias:
+        return alias
+    elif not existing_alias:
+        response = ds_client.create_alias(DirectoryId=directory_id, Alias=alias)
+        logger.info("Directory alias created")
+        logger.debug(f"create_alias_response = {json.dumps(response, default=str)}")
+        return alias
+    else:
+        error_message = f"Directory already has a different alias.  Use '{existing_alias}' for the 'DirectoryAlias' CloudFormation parameter."
+        raise Exception(error_message)
```

Comment:
No practical difference to the code, but I'd raise something like `ValueError` here rather than the base `Exception`.